### PR TITLE
Cs login pt12 pt14

### PR DESCRIPTION
**Pull request by csonparote**
```
Feature: Validate new login 


@FunctionalTest
Scenario: Validate new login features 

#Given Navigate to URL on <browser>
Then Login To Platform
And logout


#Examples:
#|browser|
#|chrome	|


@APF_Login_InvalidUser
Scenario: Login with invalid user name
Given User is on home page
When User clicks login button
And User enters invalid user name and valid password
Then Validate that error is displayed for invalid user name

@APF_Login_InvalidPassword
Scenario: Login with invalid password
Given User is on home page
When User clicks login button
And User enters valid user name and invalid password
Then Validate that error is displayed for invalid password

@APF_Login_InvalidDetails
Scenario: Login with invalid password
Given User is on home page
When User clicks login button
And User enters invalid details
Then Validate that error is displayed for invalid details
```

### CODE REVIEW

File: platform/src/test/java/platformstepdefinition/PF_Login.java [hunk 1]
```diff
@@ -77,28 +78,67 @@ public void navigate_to_home_page() throws Throwable {
...
+		loginPage.setInvalidUserData("password");
+	}
+
+	@Then("^Validate that error is displayed for invalid details$")
+	public void validate_that_error_is_displayed_for_invalid_details() throws Throwable {
+		// throw new PendingException();
+
+		NewBaseClass newBaseClass = new NewBaseClass();
+		LoginPage loginPage = new LoginPage(driver);
+		newBaseClass.validateErrorMessage(loginPage.invalidDataErrorText, "メールアドレスは、メールアドレス形式で入力してください。");
```
Comment (Sopan-Patil) [inline platform/src/test/java/platformstepdefinition/PF_Login.java]: content should not use. Please use it through excel.

File: platform/src/test/java/platformstepdefinition/PF_Login.java [hunk 2]
```diff
@@ -77,28 +78,67 @@ public void navigate_to_home_page() throws Throwable {
...
+		loginPage.setInvalidUserData("user");
+
+	}
 
+	@Then("^Validate that error is displayed for invalid password$")
+	public void validate_that_error_is_displayed_for_invalid_password() throws Throwable {
+		// throw new PendingException();
+		NewBaseClass newBaseClass = new NewBaseClass();
+		LoginPage loginPage = new LoginPage(driver);
+		newBaseClass.validateErrorMessage(loginPage.invalidDataErrorText, "メールアドレスまたEiDまたはパスワードをご確認してください。");
```
Comment (Sopan-Patil) [inline platform/src/test/java/platformstepdefinition/PF_Login.java]: content should not use. Please use it through excel.

File: platform/src/test/java/platformstepdefinition/PF_Login.java [hunk 3]
```diff
@@ -77,28 +78,67 @@ public void navigate_to_home_page() throws Throwable {
...
+		LoginPage loginPage = new LoginPage(driver);
+		loginPage.clickLoginButton();
+	}
+
+	@Then("^Validate that error is displayed for invalid user name$")
+	public void validate_that_error_is_displayed_for_invalid_user_name() throws Throwable {
+		// throw new PendingException();;
+		NewBaseClass newBaseClass = new NewBaseClass();
+		LoginPage loginPage = new LoginPage(driver);
+		newBaseClass.validateErrorMessage(loginPage.invalidDataErrorText, "メールアドレスは、メールアドレス形式で入力してください。");
```
Comment (Sopan-Patil) [inline platform/src/test/java/platformstepdefinition/PF_Login.java]: content should not use. Please use it through excel.

**Review by Sopan-Patil — CHANGES_REQUESTED** [review]

File: platform/src/test/java/platformstepdefinition/PF_Login.java (hunk 1 above)
Comment (csonparote) [reply platform/src/test/java/platformstepdefinition/PF_Login.java]: Validating error message as done with Order module.
Seperate script to be written for validating strings

File: platform/src/test/java/platformstepdefinition/PF_Login.java (hunk 2 above)
Comment (csonparote) [reply platform/src/test/java/platformstepdefinition/PF_Login.java]: Validating error message as done with Order module.
Seperate script to be written for validating strings

File: platform/src/test/java/platformstepdefinition/PF_Login.java (hunk 3 above)
Comment (csonparote) [reply platform/src/test/java/platformstepdefinition/PF_Login.java]: Validating error message as done with Order module.
Seperate script to be written for validating strings

**Comment by csonparote** [comment]
Done suggested changes. 
Please review changed files

File: platform/src/main/java/platform/pageobjects/Authentication/LoginPage.java [hunk 4]
```diff
@@ -225,4 +229,63 @@ public void navigateToHomePage() throws Exception {
...
+
+	public void setInvalidUserData(String invalid) throws Exception {
+
+		ExcelUtil excel = new ExcelUtil();
+		excel.setExcelFile("NewTestData.xlsx", "NewUser");
+
+		CommonFunctions.isElementVisible(emailtextfield);
+		// String[] userData;
+
+		// userData = XLHandler.readexcel("User", "NewTestData.xlsx");
```
Comment (Sopan-Patil) [inline platform/src/main/java/platform/pageobjects/Authentication/LoginPage.java]: remove commneted code

File: platform/src/test/java/platformstepdefinition/PF_Login.java [hunk 5]
```diff
@@ -77,28 +78,77 @@ public void navigate_to_home_page() throws Throwable {
...
+		LoginPage loginPage = new LoginPage(driver);
+		loginPage.clickLoginButton();
+	}
+
+	@Then("^Validate that error is displayed for invalid user name$")
+	public void validate_that_error_is_displayed_for_invalid_user_name() throws Throwable {
+		// throw new PendingException();;
+		NewBaseClass newBaseClass = new NewBaseClass();
+		// LoginPage loginPage = new LoginPage(driver);
+
```
Comment (Sopan-Patil) [inline platform/src/test/java/platformstepdefinition/PF_Login.java:97]: Remove unwanted code

**Review by Sopan-Patil — CHANGES_REQUESTED** [review]

File: platform/src/main/java/platform/pageobjects/Authentication/LoginPage.java (hunk 4 above)
Comment (csonparote) [reply platform/src/main/java/platform/pageobjects/Authentication/LoginPage.java]: done

File: platform/src/test/java/platformstepdefinition/PF_Login.java (hunk 5 above)
Comment (csonparote) [reply platform/src/test/java/platformstepdefinition/PF_Login.java:97]: done

**Comment by csonparote** [comment]
Done above changes.
Please check in changed files

**Review by Sopan-Patil — APPROVED** [review]

**Pull request merged**